### PR TITLE
Experimental client for horizontal pod autoscaler.

### PR DESCRIPTION
**Pull request by jszczepkowski**
Implementation of experimental client for horizontal pod autoscaler. Placeholder implementation for controller for horizontal pod autoscaler.

### CODE REVIEW

File: pkg/expapi/testapi/testapi.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+package testapi
+
+import (
+	"strings"
+
+	"k8s.io/kubernetes/pkg/expapi/latest"
+)
+
+// Returns the appropriate path for the given prefix (watch, proxy, redirect, etc), resource, namespace and name.
+// For ex, this is of the form:
```
Comment (davidopp) [inline pkg/expapi/testapi/testapi.go]: spell out "example"

File: pkg/expapi/testapi/testapi.go [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+// /expapi/v1/watch/namespaces/foo/pods/pod0 for v1.
```
Comment (davidopp) [inline pkg/expapi/testapi/testapi.go]: I think you mean /experimental/ not /expapi/ at least based on what the code is doing

File: pkg/client/horizontalpodautoscaler.go [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+}
+
+// Get takes the name of the horizontalPodAutoscaler, and returns the corresponding HorizontalPodAutoscaler object, and an error if it occurs
+func (c *horizontalPodAutoscalers) Get(name string) (result *expapi.HorizontalPodAutoscaler, err error) {
+	result = &expapi.HorizontalPodAutoscaler{}
+	err = c.r.Get().Namespace(c.ns).Resource("horizontalPodAutoscalers").Name(name).Do().Into(result)
+	return
+}
+
+// Delete takes the name of the horizontalPodAutoscaler, and returns an error if one occurs
```
Comment (davidopp) [inline pkg/client/horizontalpodautoscaler.go]: I noticed that for Delete, Create, and Update you didn't mention that the functions actually delete, create, and update the corresponding HorizontalPodAutoscaler. Is this a standard Go rule for writing function comments? If not, please mention these aspects of the functions.

File: pkg/client/horizontalpodautoscaler.go [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+	Get(name string) (*expapi.HorizontalPodAutoscaler, error)
+	Delete(name string, options *api.DeleteOptions) error
+	Create(horizontalPodAutoscaler *expapi.HorizontalPodAutoscaler) (*expapi.HorizontalPodAutoscaler, error)
+	Update(horizontalPodAutoscaler *expapi.HorizontalPodAutoscaler) (*expapi.HorizontalPodAutoscaler, error)
+	Watch(label labels.Selector, field fields.Selector, resourceVersion string) (watch.Interface, error)
+}
+
+// horizontalPodAutoscalers implements HorizontalPodAutoscalersNamespacer interface
+type horizontalPodAutoscalers struct {
+	r  *ExperimentalClient
```
Comment (davidopp) [inline pkg/client/horizontalpodautoscaler.go]: call this "client" instead of "r" ?

File: pkg/expapi/testapi/testapi.go (hunk 1 above)
Comment (jszczepkowski) [inline pkg/expapi/testapi/testapi.go]: Done.

File: pkg/expapi/testapi/testapi.go (hunk 2 above)
Comment (jszczepkowski) [inline pkg/expapi/testapi/testapi.go]: Done.

File: pkg/client/horizontalpodautoscaler.go (hunk 3 above)
Comment (jszczepkowski) [inline pkg/client/horizontalpodautoscaler.go]: Done.

File: pkg/client/horizontalpodautoscaler.go (hunk 4 above)
Comment (jszczepkowski) [inline pkg/client/horizontalpodautoscaler.go]: Done.

**Comment by jszczepkowski** [comment]
@davidopp All comments are applied, PTAL.

**Comment by davidopp** [comment]
LGTM

**Comment by jszczepkowski** [comment]
Part of #12087.

**Pull request merged**